src/azul/service/manifest_service.py

```diff
@@ -703,10 +704,54 @@ def write_to(self, output: IO[str]) -> Optional[str]:
             url = furl(config.service_endpoint(),
                        path=f'/repository/files/{uuid}',
                        args=dict(version=version, catalog=self.catalog))
+            name = self._validate(name)
             output.write(f'url={self._option(url.url)}\n'
                          f'output={self._option(name)}\n\n')
         return None
 
+    @classmethod
+    def _validate(cls, path: str):
+        """
+        >>> f = CurlManifestGenerator._validate
+        >>> f('foo/bar/\\x1F/file')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Invalid file name 'foo/bar/\\x1f/file'
+
+        >>> f('foo/bar/\\/file')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Invalid file name 'foo/bar/\\\\/file'
+
+        >>> f('foo/bar/COM6/file')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Invalid file name 'foo/bar/COM6/file'
+
+        >>> f('foo/bar?/fi*le|')
+        'foo/bar_/fi_le_'
+
+        >>> f('foo/bar/file.fastqgz')
+        'foo/bar/file.fastqgz'
+        """
+        msg = f'Invalid file name {path!r}'
+        for character in path:
+            reject(unicodedata.category(character) == 'Cc'
+                   or character == u'\\',
+                   msg)
+        path = re.sub(r'[<>:"|?*]', '_', path).strip()
+        re_name = re.match(r'^(\.?[^./ ])([^/]*[^./ ])?(/(\.?[^./ ])([^/]*[^./ ])?)*$',
```

Comment:
Personal preference, but I think it might be clearer if we dropped the `^$` anchors and used [fullmatch](https://docs.python.org/3.8/library/re.html#re.fullmatch)

Reply:
I was fallowing the [instructions from the ticket](https://github.com/DataBiosphere/azul/issues/2398#issue-733576416) item 4. Additionally, I don't see a `^$` did you mean something else?

Reply:
I meant the `^` (start of string) and `$` (end of string) anchors, separately, at the start and end of the regex, respectively. I don't interpret my suggestion as contradicting the ticket instructions, but if you do, then it's fine as-is.